pyuvdata/uvflag.py

```diff
@@ -127,8 +134,11 @@ def __init__(self, input, mode='metric', copy_flags=False, waterfall=False, hist
                     self.flag_array = np.zeros_like(input.flag_array)
                 elif self.mode == 'metric':
                     self.metric_array = np.zeros_like(input.flag_array).astype(np.float)
+        else:
+            raise ValueError('input to UVFlag.__init__ must be one of: list, tuple, '
+                             'string, UVData, or UVCal.')
 
-        if not isinstance(input, str):
+        if issubclass(input.__class__, (UVData, UVCal)):
```

Comment:
It looks like this line logic was changed but I can't seem to find a test that covers this change.  Did I miss a test?

Reply:
New test: `test_init_list_files_weights`